### PR TITLE
Add git-lfs support to the git client

### LINKED ISSUE #180: Install git-lfs
I was caught off guard when the first application I published that was built using `cimg/node:lts`, had users complaining about errors for missing images. I assumed anything CI related would include git and git-lfs so as to pull my entire code base.

### PR DESCRIPTION
**Pull request by Peter-Darton-i2**
TL;DR: `apt-get install git-lfs`
Closes #180

## What is it?

Git's answer to the problems of having large files in a git repository is git-lfs.

## Why do we care?

Since git is core to how CircleCI works, and since git-lfs is core to how git handles large files, many customers will need this.
Any repo might be using git-lfs; it's typically installed "by default" in development environments etc such that user's might not even be aware of its presence.
Some git repo providers (e.g. GitHub) even force users to use it for files over a certain size.

However, IF a CircleCI build does a `checkout` on a repository that contains git-lfs files AND the git client is _missing_ support for git-lfs THEN it does not fail, it just runs the build with a corrupted workspace, with the "lfs files" being left as (small) references to git-lfs content instead of being the actual files they should be.
Silent failures are bad - unless the customer's build also includes some form of test to verify the integrity of the data it got from git, it could result in a corrupted build result being published.

## How do we fix it?

We install LFS support into the git client.
For consistency, as these docker images are already using the "latest-stable" release of the git core from its private package area (done in #113), we use the equivalent "latest-stable" version of git-lfs from [its private package area](https://github.com/git-lfs/git-lfs/blob/main/INSTALLING.md) too.

## What's the impact?

Git-lfs adds approximately 15 megabytes to the image, which is a 1.4% increase overall.

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -23,15 +23,15 @@ RUN echo 'APT::Get::Assume-Yes "true";' > /etc/apt/apt.conf.d/90circleci && \
 	&& \
 	locale-gen en_US.UTF-8 && \
 	rm -rf /var/lib/apt/lists/* && \
-
+	\
```
Comment (Peter-Darton-i2) [inline Dockerfile.template:26]: This change isn't core to this PR, but the docker build was issuing a warning about unsafe blank lines and the fact that the warning would turn into error in later docker releases.

File: Dockerfile.template [hunk 2]
```diff
@@ -89,7 +89,10 @@ RUN noInstallRecommends="" && \
 		vim \
 		wget \
 		zip && \
+	# get the semi-official latest-stable git instead of using the old(er) version from the ubuntu distro
```
Comment (Peter-Darton-i2) [inline Dockerfile.template:92]: Added a comment explaining why #113 made this change

File: Dockerfile.template [hunk 3]
```diff
@@ -89,7 +89,10 @@ RUN noInstallRecommends="" && \
...
 	add-apt-repository ppa:git-core/ppa && apt-get install -y git && \
+	# get the semi-official latest-stable git-lfs too
```
Comment (Peter-Darton-i2) [inline Dockerfile.template:94]: ...so that it'd be more obvious to a reader why we're not simply using the git-lfs from the Ubuntu core distro here too.

File: Dockerfile.template [hunk 4]
```diff
@@ -89,7 +89,10 @@ RUN noInstallRecommends="" && \
...
+	curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash && apt-get install -y git-lfs && \
```
Comment (Peter-Darton-i2) [inline Dockerfile.template:95]: This added line is the key change in this PR.
We add the packagecloud repo and then apt-get install git-lfs.

The only thing I'm not clear on is whether or not `$noInstallRecommends` should be provided as an argument to the apt-get command - that argument is used for most of the package installs, but the line that installs git does NOT include it. I followed git's example, but it might be that git's installation line is incorrect as it's inconsistent...

**Comment by JalexChen** [comment]
Hi @Peter-Darton-i2  - thanks for the PR. Running this right now but otherwise looks good to me

Some notes:

- This image is intended to be as lightweight as possible, so no install recommends is generally preferred vs the git example
- we would also ideally like to pin the version of git-lfs, but it doesn't seem possible if 18.04 is also being built so leaving this as a comment for the future when it's more addressable

**Comment by JalexChen** [comment]
Hi Peter - 

Agreeing with you on both parts. both should be saved for a later date when it's more addressable

File: Dockerfile.template (hunk 4 above)
Comment (JalexChen) [reply Dockerfile.template:95]: LGTM

**Review by JalexChen — APPROVED** [review]
